### PR TITLE
Users, Invites, Workspaces and Members API

### LINKED ISSUE #87: NodeSDK: User and Invite API
Introducing support for Users and User Invite APIs

### PR DESCRIPTION
**Pull request by csgulati09**
**Title:** User , Invite , Workspace and Workspace Members API

**Description:**
- Added admin.users
- Added admin.users.invites 
- Added admin.workspace
- Added admin.workspace.members
- All method supported
- Config changes to support .get and .delete methods 

**Related Issues:**
Closes #87  #118

### CODE REVIEW

File: src/apis/admin.ts [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+        this.workspaces= new Workspaces(client);
+    }
+}
+
+
+export class Users extends ApiResource {
+
+    invites: Invites
+
+    constructor(client: any) {
```
Comment (csgulati09) [inline src/apis/admin.ts:233]: Can remove the constructor and implement it in this way 

`invites: Invites = new Invites(this.client);`

File: src/apis/admin.ts [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const userId = _body.userId;
+        delete body.userId;
+        if (params) {
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.put<any>(`/admin/users/${userId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:278]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+        _body: UsersDeleteParams,
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const userId = _body.userId;
+        if (params){
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.deleteMethod<any>(`/admin/users/${userId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:292]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.deleteMethod<any>(`/admin/users/invites/${inviteId}`, { body, ...opts });
+        return response;
+    }
+
+}
+
+export class Workspaces extends ApiResource {
+    users: Member
```
Comment (csgulati09) [inline src/apis/admin.ts:359]: Initialisation change here as well

File: src/apis/admin.ts [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+        _body: UserInviteDeleteParams,
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const inviteId = _body.inviteId;
+        if (params){
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.deleteMethod<any>(`/admin/users/invites/${inviteId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:352]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+        _body: WorkspacesDeleteParams,
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const workspaceId = _body.workspaceId;
+        if (params){
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.deleteMethod<any>(`/admin/workspaces/${workspaceId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:430]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+        _body: WorkspaceMemberAddParams,
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const workspaceId = body.workspaceId;
+        if (params) {
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.post<any>(`/admin/workspaces/${workspaceId}/users`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:447]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+        params?: ApiClientInterface,
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const workspaceId = body.workspaceId;
+        const userId = body.userId;
+        if (params){
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.deleteMethod<any>(`/admin/workspaces/${workspaceId}/users/${userId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:491]: Output type has to be `<any>` or we can make an interface for it as well?

File: src/apis/admin.ts [hunk 9]
```diff
@@ -0,0 +1,509 @@
...
+        opts?: RequestOptions
+    ): APIPromise<any> {
+        const body = _body;
+        const workspaceId = body.workspaceId;
+        const userId = _body.userId;
+        delete body.workspaceId;
+        if (params) {
+            this.client.customHeaders = { ...this.client.customHeaders, ...createHeaders({ ...params }) }
+        }
+        const response = this.put<any>(`/admin/workspaces/${workspaceId}/users/${userId}`, { body, ...opts });
```
Comment (csgulati09) [inline src/apis/admin.ts:506]: Output type has to be `<any>` or we can make an interface for it as well?

**Review by VisargD — APPROVED** [review]

**Pull request merged**